lib/compat/wordpress-6.0/block-patterns.php

```diff
@@ -38,4 +32,18 @@ function gutenberg_register_remote_theme_patterns() {
 	}
 }
 
-add_action( 'init', 'gutenberg_register_remote_theme_patterns' );
+add_action(
+	'current_screen',
+	function( $current_screen ) {
+		$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+		$theme_has_support  = WP_Theme_JSON_Resolver_Gutenberg::theme_has_support();
+		if ( ! get_theme_support( 'core-block-patterns' ) || ! $should_load_remote || ! $theme_has_support ) {
```

Comment:
Written this way the code always evaluates all three conditions before checking them. We can return early after checking each individual one.

Reply:
I updated it.